### ISSUE TITLE
Update dependencies

### ISSUE THREAD

**Issue by mitchhentges** [mozilla/sccache#1110]
Running [`cargo outdated -R`](https://github.com/kbknapp/cargo-outdated) shows:
```
Name             Project           Compat  Latest             Kind         Platform
----             -------           ------  ------             ----         --------
ar               0.8.0             ---     0.9.0              Normal       ---
assert_cmd       1.0.8             ---     2.0.4              Development  ---
blake3           0.3.8             ---     1.3.1              Normal       ---
clap             2.34.0            ---     3.0.14             Normal       ---
directories      3.0.2             ---     4.0.1              Normal       ---
env_logger       0.8.4             ---     0.9.0              Normal       ---
futures-locks    0.6.0             ---     0.7.0              Normal       ---
hmac             0.10.1            ---     0.12.0             Normal       ---
hyperx           0.13.2            ---     1.4.0              Normal       ---
jsonwebtoken     7.2.0             ---     8.0.1              Normal       ---
md-5             0.9.1             ---     0.10.0             Normal       ---
nix              0.19.1            ---     0.23.1             Normal       ---
predicates       1.0.8             ---     2.1.1              Development  ---
rand             0.7.3             ---     0.8.4              Normal       ---
sha-1            0.9.8             ---     0.10.0             Normal       ---
sha2             0.9.9             ---     0.10.1             Normal       ---
syslog           5.0.0             ---     6.0.1              Normal       ---
tokio-util       0.6.9             ---     0.7.0              Normal       ---
untrusted        0.7.1             ---     0.9.0              Normal       ---
version-compare  0.0.11            ---     0.1.0              Normal       ---
zstd             0.6.1+zstd.1.4.9  ---     0.10.0+zstd.1.5.2  Normal       ---
```

Let's bump these dependencies.
There'll probably be some breaking changes, so Rust experience will likely be necessary as part of this work.

**Comment by jschwe** [comment]
I would take a look at this. What is the policy about bumping the MSRV, if updating dependencies bumps their MSRV above ours?

**Comment by jschwe** [comment]
I created a PR (#1113) for the easy dependencies. I have ignored the crypto crates, clap and tokio for now, since I expect more work there.

The largest open question for me is: How far can we bump the MSRV? Is there a hard requirement due to packaging for some distribution?

A short check of the crates not covered by my PR:
- blake3: upgrade to 1.0 requires MSRV of 1.51 (const generics). the versions up to 1.3 don't document another increase, but it looks like they don't test 1.51 in CI, so this might be too low.
- z-std seems to require a MSRV bump also for const generics ( so probably also 1.51, but I couldn't find anything official )
- jsonwebtoken 8 requires rust 1.56 (2021 edition), which is probably too recent?
- syslog has an MSRV of 1.53
- assert_cmd + ppredicates : MSRV 1.54, but 1.48 seems to work for the current versions.
- clap 3 (MSRV 1.54). (But I don't think updating to v3 has to be done now, does it?)
- tokio: MSRV is currently 1.49.

**Comment by sylvestre** [comment]
great work, bravo :)

**Comment by mitchhentges** [comment]
> The largest open question for me is: How far can we bump the MSRV? Is there a hard requirement due to packaging for some distribution?

Good question, let's wait on bumping the MSRV _at least_ until after the next release.
In the meantime, I'm going to do some digging into what restrictions we have in bumping it.

Thanks for the patch!

**Comment by LovecraftianHorror** [comment]
It looks like after #1137 gets merged the potentially non-trivial updates are just

- [ ] `clap` 2.34.0 -> 3.1.6
- [ ] `tokio-util` 0.6.9 -> 0.7.0

unless I'm missing another PR

I can take on the clap upgrade unless someone else is already working on it. It will require an MSRV bump, but that shouldn't be a problem after #1137 gets merged

**Comment by sylvestre** [comment]
we fixed most of them by now.
